test/hierarchies.cc

```diff
@@ -219,3 +222,124 @@ TEST(lin_reg_uni_hierarchy, misc) {
+  hier->write_state_to_proto(clusval);
+  hier2->write_state_to_proto(clusval2);
+
+  ASSERT_TRUE(clusval->DebugString() != clusval2->DebugString());
+}
+
+TEST(mfahierarchy, sample_given_data) {
+  auto hier = std::make_shared<MFAHierarchy>();
+  bayesmix::MFAPrior prior;
+  Eigen::VectorXd mutilde(4);
+  mutilde << 3.0, 3.0, 4.0, 1.0;
+  bayesmix::Vector mutilde_proto;
+  bayesmix::to_proto(mutilde, &mutilde_proto);
+  int q = 2;
+  double phi = 1.0;
+  double alpha0 = 5.0;
+  Eigen::VectorXd beta(4);
+  beta << 3.0, 3.0, 2.0, 2.1;
+  bayesmix::Vector beta_proto;
+  bayesmix::to_proto(beta, &beta_proto);
+  *prior.mutable_fixed_values()->mutable_mutilde() = mutilde_proto;
+  prior.mutable_fixed_values()->set_phi(phi);
+  prior.mutable_fixed_values()->set_alpha0(alpha0);
+  prior.mutable_fixed_values()->set_q(q);
+  *prior.mutable_fixed_values()->mutable_beta() = beta_proto;
+  hier->get_mutable_prior()->CopyFrom(prior);
+  hier->initialize();
+
+  Eigen::RowVectorXd datum1(4);
+  Eigen::RowVectorXd datum2(4);
+  datum1 << 1.0, 2.0, 3.0, 4.0;
+  datum2 << 2.0, 4.0, 6.0, 8.0;
+
+  auto hier2 = hier->clone();
+  hier2->add_datum(0, datum1, false);
+  hier2->add_datum(1, datum2, false);
+  hier2->sample_full_cond();
+  bayesmix::AlgorithmState out;
+  bayesmix::AlgorithmState::ClusterState* clusval = out.add_cluster_states();
+  bayesmix::AlgorithmState::ClusterState* clusval2 = out.add_cluster_states();
+  hier->write_state_to_proto(clusval);
+  hier2->write_state_to_proto(clusval2);
+
+  ASSERT_TRUE(clusval->DebugString() != clusval2->DebugString());
+}
+
+TEST(mfahierarchy, sample_full_cond) {
+  auto hier = std::make_shared<MFAHierarchy>();
+  bayesmix::MFAPrior prior;
+  Eigen::VectorXd mutilde(10);
+  mutilde << 1.1, 2.2, 3.1, 4.1, 4.8, 6.1, 6.9, 8.2, 9.0, 10.2;
+  bayesmix::Vector mutilde_proto;
+  bayesmix::to_proto(mutilde, &mutilde_proto);
+  int q = 5;
+  double phi = 1.0;
+  double alpha0 = 2.0;
+  Eigen::VectorXd beta(10);
+  beta << 1, 1, 1, 1, 1, 1, 1, 1, 1, 1;
+  bayesmix::Vector beta_proto;
+  bayesmix::to_proto(beta, &beta_proto);
+  *prior.mutable_fixed_values()->mutable_mutilde() = mutilde_proto;
+  prior.mutable_fixed_values()->set_phi(phi);
+  prior.mutable_fixed_values()->set_alpha0(alpha0);
+  prior.mutable_fixed_values()->set_q(q);
+  *prior.mutable_fixed_values()->mutable_beta() = beta_proto;
+  hier->get_mutable_prior()->CopyFrom(prior);
+  hier->initialize();
+
+  Eigen::MatrixXd data =
+      bayesmix::read_eigen_matrix("../examples/mfa_hierarchy/in/data.csv");
+
+  for (unsigned int i = 0; i < data.rows(); i++) {
+    Eigen::RowVectorXd row = data.row(i);
+    hier->add_datum(i, row, false);
+  }
+
+  Eigen::MatrixXd musamplato(500, 10);
+  Eigen::MatrixXd psisamplato(500, 10);
+
+  for (unsigned int i = 0; i < 500; i++) {
+    hier->sample_full_cond();
+    MFA::State stato = hier->get_state();
+    psisamplato.row(i) = stato.psi;
+  }
+
+  //bayesmix::write_matrix_to_file(
```

Comment:
remove the commented lines

Reply:
also it feels like this test is not checking anything. Maybe reduce the number of iterations to one or two and perform an actual check